libraries/chain_kv/include/b1/session/rocks_data_store.hpp

```diff
@@ -0,0 +1,684 @@
+   auto value     = std::string{};
+   return m_db->KeyMayExist(m_read_options, column_family_(), key_slice, &value);
+}
+
+template <typename Allocator>
+void rocks_data_store<Allocator>::erase(const shared_bytes& key) {
+   if (!m_db) {
+      return;
+   }
+
+   auto key_slice = rocksdb::Slice{ reinterpret_cast<const char*>(key.data()), key.length() };
+   auto status    = m_db->Delete(m_write_options, column_family_(), key_slice);
+}
+
+template <typename Allocator>
+void rocks_data_store<Allocator>::clear() {
+   // TODO:
+}
+
+template <typename Allocator>
+template <typename Iterable, typename Other_allocator>
+const std::pair<std::vector<key_value>, std::unordered_set<shared_bytes>>
+rocks_data_store<Allocator>::read_(const Iterable& keys, const std::shared_ptr<Other_allocator>& a) const {
+   if (!m_db) {
+      return {};
+   }
+
+   auto not_found  = std::unordered_set<shared_bytes>{};
+   auto key_slices = std::vector<rocksdb::Slice>{};
+
+   for (const auto& key : keys) {
+      key_slices.emplace_back(reinterpret_cast<const char*>(key.data()), key.length());
+      not_found.emplace(key);
+   }
+
+   auto values = std::vector<std::string>{};
+   values.reserve(key_slices.size());
+   auto status = m_db->MultiGet(m_read_options, { key_slices.size(), column_family_() }, key_slices, &values);
+
+   auto kvs = std::vector<key_value>{};
+   kvs.reserve(key_slices.size());
+
+   for (size_t i = 0; i < values.size(); ++i) {
+      if (status[i].code() != rocksdb::Status::Code::kOk) {
+         continue;
+      }
+
+      not_found.erase(make_shared_bytes(key_slices[i].data(), key_slices[i].size()));
+      kvs.emplace_back(make_kv(key_slices[i].data(), key_slices[i].size(), values[i].data(), values[i].size(), a));
+   }
+
+   return { std::move(kvs), std::move(not_found) };
+}
+
+// Reads a batch of keys from rocksdb.
+//
+// \tparam Iterable Any type that can be used within a range based for loop and returns shared_bytes instances in its
+// iterator. \param keys An Iterable instance that returns shared_bytes instances in its iterator. \returns An std::pair
+// where the first item is list of the found key_values and the second item is a set of the keys not found.
+template <typename Allocator>
+template <typename Iterable>
+const std::pair<std::vector<key_value>, std::unordered_set<shared_bytes>>
+rocks_data_store<Allocator>::read(const Iterable& keys) const {
+   return read_(keys, m_allocator);
+}
+
+// Writes a batch of key_values to rocksdb.
+//
+// \tparam Iterable Any type that can be used within a range based for loop and returns key_value instances in its
+// iterator. \param key_values An Iterable instance that returns key_value instances in its iterator.
+template <typename Allocator>
+template <typename Iterable>
+void rocks_data_store<Allocator>::write(const Iterable& key_values) {
+   if (!m_db) {
+      return;
+   }
+
+   auto batch = rocksdb::WriteBatch{ 1024 * 1024 };
```

Comment:
It was mainly chosen because I thought blocks were at most 1MB.  So I reserved space for 1MB.  Performance testing determine if a parameter is required or not.